test/Main.hs

```diff
@@ -686,14 +689,19 @@ streamOperations (stream, list, len) = do
         it "last empty" $ A.last stream `shouldReturn` Nothing
         it "maximum empty" $ A.maximum stream `shouldReturn` Nothing
         it "minimum empty" $ A.minimum stream `shouldReturn` Nothing
+        it "null empty" $ A.null stream `shouldReturn` True
+        it "tail empty" $ A.tail stream >>= 
+            \t -> if isNothing t then return () else expectationFailure "failed"
```

Comment:
Why not write it as:

```
it "tail empty" $ A.tail stream `shouldReturn` Nothing
```

Reply:
because `Maybe (t m a)` doesn't have a `Show` instance, which is required by `shouldReturn`

Reply:
Ah, ok. In that case you can write as:

```
it "tail empty" $ A.tail stream >>= maybe (return ()) (expectationFailure "failed")
```